src/state.ml

```diff
@@ -393,9 +393,9 @@ let channel_of_keyid keyid s =
             Some (channel, set)
         | _ -> None)
 
-let transition_to_established t =
-  match (t.control_crypto, t.state) with
-  | (Tls_auth _ | Tls_crypt _), Client (Handshaking _) ->
+let transition_to_established t control_crypto =
+  match (control_crypto, t.state) with
+  | #control_tls, Client (Handshaking _) ->
```

Comment:
Thinking out loud: we don't ever really deconstruct or use `control_crypto`. But we pass it on because we can sort of ensure it's a tls client (or server) this way.

Reply:
turns out, we can as well just drop it. it's purpose right now is to specify at the type level that `transition_to_established` will only be executed with Tls_auth / Tls_crypt

Reply:
Yes, we can drop it. It got me thinking if we should have again a `Client_static` that only allows the client states that are relevant. `Handshaking` and `Rekeying` are not valid states for a static client. On the other hand this might reintroduce some pain points. I will do some experiments.